src/inspect_ai/analysis/beta/_dataframe/samples/table.py

```diff
@@ -356,26 +356,27 @@ def _read_samples_df_serial(
         details_table.drop_duplicates(
             f"{detail.name}_id", keep="first", inplace=True, ignore_index=True
         )
-        samples_table = details_table.merge(
-            samples_table,
-            on=SAMPLE_ID,
-            how="left",
-            suffixes=(f"_{detail.name}", SAMPLE_SUFFIX),
-        )
+        if len(details_table) > 0:
```

Comment:
I think we need an `else` here so that sets `samples_table = details_table` which will then yield an empty result for the entire call, as if there are no detail targets (events or messages) we want the resulting dataframe to be empty.

Reply:
I am wrong about this -- in the case where there are no samples there are by definition no details. In the case t that `samples > 0` there are always details (at least one message or event per sample). So all good!